test/inject/inject_test.go

```diff
@@ -0,0 +1,57 @@
+package inject
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/linkerd/linkerd2/testutil"
+)
+
+var TestHelper *testutil.TestHelper
+
+func TestMain(m *testing.M) {
+	TestHelper = testutil.NewTestHelper()
+	os.Exit(m.Run())
+}
+
+func TestCliInject(t *testing.T) {
+	resourceFolder := filepath.Join("testdata", "resources")
+
+	t.Run("read from files", func(t *testing.T) {
+		testCases := []struct {
+			resource     string
+			resourceFile string
+			expectedFile string
+		}{
+			{resource: "nginx", resourceFile: filepath.Join(resourceFolder, "nginx.yaml"), expectedFile: filepath.Join("expected", "injected_nginx.yaml")},
+			{resource: "redis", resourceFile: filepath.Join(resourceFolder, "db/redis.yaml"), expectedFile: filepath.Join("expected", "injected_redis.yaml")},
+		}
```

Comment:
Hey @ihcsim, thanks for putting this change together! It's a great new feature. As far as testing goes, the `test/` directory is intended for integration tests, which are used to test linkerd2 installs end-to-end. Since integration tests require building all of the docker images first, we only run integration tests after each master commit.

The tests that you're adding only tests the CLI in isolation (no linkerd2 install required), so they could instead be written as unit tests and added to `cli/cmd/inject_test.go`. The main advantage of this is that they'll be run as part of the unit test suite, which is run on every commit and branch, so they'll help to catch problems much sooner than if they're only run as part of the integration tests.

Can I trouble you to move these tests to `cli/cmd/inject_test.go`? You can put the input YAML files into a subdirectory within `cli/cmd/testdata/`. In fact that directory already has other YAML files that we use to vet the inject functionality on existing Kubernetes configs.